syntax = "proto3";

package e_client;

service IMAPClientService {
    rpc FetchEmails(LimitsRequest) returns (FetchEmailsResponse);
    rpc FetchEmailInfo(IdEmailRequest) returns (FetchEmailInfoResponse);
}

message LimitsRequest {
    int32 offset = 1;
    int32 limit = 2;
}

message IdEmailRequest {
    int64 id = 1;
}

message SummaryEmailResponse {
    int64 id = 1;
    string sender = 2;
    string subject = 3;
    string date = 4;
}

message FetchEmailsResponse {
    repeated SummaryEmailResponse emailsList = 1;
}

message FetchEmailInfoResponse {
    string sender = 1;
    string to = 2;
    string subject = 3;
    string date = 4;
    string body = 5;
    repeated string attachments = 6;
}
